scielo_scholarly_data/standardizer.py

```diff
@@ -460,6 +461,48 @@ def document_author_for_deduplication(text: str, surname_first=True):
     return text
 
 
+def orcid_validator(text: str, type=1):
+    """
+        Função para verificar e padronizar um registro ORCID.
+
+        Parameters
+        ----------
+        text : str
+            Registro ORCID a ser validado e padronizado.
+        type : int
```

Comment:
@Rossi-Luciano acredito que este parâmetro pode ser mais específico. Aliás, podemos ter mais de um parâmetro para indicar o que queremos fazer.

Sugiro: 

1. Mudar `type` para `exec_mode` ou `mode`;
2. Usar biblioteca urllib.parse para fazer o parse de um endereço uri (veja exemplos [aqui](https://docs.python.org/3/library/urllib.parse.html);
3. Ao usar o parse indicado em 2, será possível extrair o `host` e o `path` - é melhor extraí-los, quando possível;
4. O `path` seria o alvo da validação na situação em que a opção escolhida pelo usuário for verificar o check-sum;
5. Na situação em que não é informado o enderço completo do orcid (sem http e host), e o usuário escolher validar, o conteúdo informado é que deve ser validado;
6. Deve haver um parâmetro que a pessoa indica para apenas validar o formato (e não o checksum);

Reply:
@Rossi-Luciano aqui há exemplo de como usei o urllib.parse: https://github.com/scieloorg/scielo_usage_counter/blob/d1093d088813e90863adfd31b747c781b10088e5/app/lib/logparser.py#L374

Veja o que há dentro de um objeto retornado pela função `urllib.parse.urlparse(path)`. Basicamente há um host, um protocolo, um path e parâmetros de url, se houver.